google/cloud/dataproc_spark_connect/session.py

```
@@ -450,18 +450,32 @@ def create_session_pbar():
                         create_session_pbar_thread.join()
                     DataprocSparkSession._active_s8s_session_id = None
                     DataprocSparkSession._active_session_uses_custom_id = False
-                    raise DataprocSparkConnectException(
+
+                    error_msg = (
                         f"Error while creating Dataproc Session: {e.message}"
                     )
+
+                    # Only log in environments that don't auto-display exceptions
+                    if not environment.is_colab_enterprise():
+                        logger.error(error_msg)
+
+                    raise DataprocSparkConnectException(error_msg)
```

Should we customize `DataprocSparkConnectException` behavior in https://github.com/GoogleCloudDataproc/dataproc-spark-connect-python/blob/775e4cb4a161e11581eb1f51a57f16c59ef9c97b/google/cloud/dataproc_spark_connect/exceptions.py#L26 instead?

Seems like we still want to print just message, but do we know why it is not printed? Will throwing different message for non-Colab Enteprise would work?

---

I've tried to raise a runtime error in workbench which worked. I've tried to debug the root cause, but I didn't find the magic error swallowing logic

---

customized DataprocSparkConnectException instead